argoproj-crd/Makefile

```diff
@@ -1,18 +1,16 @@
-.PHONY: apply
-apply:
-	@if ! kubectl get namespace "$$(basename $$PWD)$$(git rev-parse --short HEAD)" >/dev/null 2>&1; then \
-		kubectl create namespace "$$(basename $$PWD)$$(git rev-parse --short HEAD)"; \
-	fi
-	helm repo add argo https://argoproj.github.io/argo-helm;
-	helm --kube-context $${KUBE_CONTEXT:-"dind"} --tiller-namespace $${TILLER_NAMESPACE:-"kube-system"} upgrade --wait --install --namespace "$$(basename $$PWD)$$(git rev-parse --short HEAD)" $$(basename $$PWD)-$$(git rev-parse --short HEAD)-argo-cd argo/argo-cd;
-	helm --kube-context $${KUBE_CONTEXT:-"dind"} --tiller-namespace $${TILLER_NAMESPACE:-"kube-system"} upgrade --wait --install --namespace "$$(basename $$PWD)$$(git rev-parse --short HEAD)" -f ci.yaml $$(basename $$PWD)-$$(git rev-parse --short HEAD) .;
+RELEASE = "$$(basename $$PWD)"
+
+.PHONY: install
+install:
+	helm repo add argo https://argoproj.github.io/argo-helm
+	helm upgrade -i --wait $(RELEASE)-argo-cd argo/argo-cd
+	helm upgrade -i --wait -f ci.yaml $(RELEASE) .
 
 .PHONY: test
 test:
-	echo "no test"
+	helm test --logs $(RELEASE)
 
-.PHONY: delete
-delete:
-	helm --kube-context $${KUBE_CONTEXT:-"dind"} --tiller-namespace $${TILLER_NAMESPACE:-"kube-system"} --namespace "$$(basename $$PWD)$$(git rev-parse --short HEAD)" ls -q \
-		| xargs -I {} helm --kube-context $${KUBE_CONTEXT:-"dind"} --tiller-namespace $${TILLER_NAMESPACE:-"kube-system"} delete --purge {};
-	kubectl delete crd applications.argoproj.io appprojects.argoproj.io
+.PHONY: uninstall
+uninstall:
+	helm uninstall $(RELEASE)
```

Comment:
Fixed namespace same as `upgrade` ?
(Since it is not fixed with `upgrade`, it is natural that there is no `namespace` here either)